sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/disbursement-receipts-integration.scheduler.e2e-spec.ts

```diff
@@ -311,6 +314,99 @@ describe(
       });
     });
 
+    it("Should import disbursement receipt file and create federal and provincial awards receipts with proper awards code mappings for a part-time application when the file contains federal and provincial receipts.", async () => {
+      // Arrange
+      const application = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        undefined,
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.Completed,
+          firstDisbursementInitialValues: {
+            documentNumber: SHARED_DOCUMENT_NUMBER,
+          },
+        },
+      );
+      mockDownloadFiles(sftpClientMock, [FEDERAL_PROVINCIAL_PART_TIME_FILE]);
+      // Queued job.
+      const { job } = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processDisbursementReceipts(job);
+
+      // Assert
+      const downloadedFile = path.join(
+        process.env.ESDC_RESPONSE_FOLDER,
+        FEDERAL_PROVINCIAL_PART_TIME_FILE,
+      );
+      expect(result).toStrictEqual([
+        {
+          processSummary: [
+            `Processing file ${downloadedFile}.`,
+            `Record with document number ${SHARED_DOCUMENT_NUMBER} at line 2 inserted successfully.`,
+            `Record with document number ${SHARED_DOCUMENT_NUMBER} at line 3 inserted successfully.`,
+            `Processing file ${downloadedFile} completed.`,
+          ],
+          errorsSummary: [],
+        },
+      ]);
+      // Assert imported receipts.
+      const [firstDisbursement] =
+        application.currentAssessment.disbursementSchedules;
+      const { bpReceipt, feReceipt } = await getReceiptsForAssert(
+        firstDisbursement.id,
+      );
+      // Assert federal receipt.
+      // Header details.
+      expect(feReceipt).toEqual(
+        expect.objectContaining({
+          batchRunDate: "2024-01-30",
+          fileDate: "2024-01-31",
+          sequenceNumber: 3228,
+        }),
+      );
+      // Document number.
+      expect(feReceipt.disbursementSchedule.documentNumber).toBe(
+        SHARED_DOCUMENT_NUMBER,
+      );
+      // Disbursed loan amount.
+      expect(feReceipt.totalDisbursedAmount).toBe(3673);
+      const feReceiptAwards = getExpectedAwardsFromReceiptValues(
+        feReceipt.disbursementReceiptValues,
+      );
+      // Disbursed grants.
+      // XYZ is a non-translated code that must be imported as it is.
+      expect(feReceiptAwards).toStrictEqual({
+        CSGF: 199,
+        CSGT: 299,
+        CSGD: 399,
+        CSGP: 499,
+        XYZ: 444,
```

Comment:
what is this XYZ? I cant understand can you please explain?

Reply:
XYZ is a non-translated code that must be imported as it is. This part of the code for the federal receipt is the same from previous test cases where the values for the federal receipt awards are acquired for `feReceiptAwards` via method `getExpectedAwardsFromReceiptValues` with input `feReceipt.disbursementReceiptValues`.